GVFS/GVFS/CommandLine/CloneVerb.cs

```diff
@@ -609,14 +609,29 @@ private void CheckNotInsideExistingRepo(string normalizedEnlistmentRootPath)
             }
 
             // Prepare the working directory folder for GVFS last to ensure that gvfs mount will fail if gvfs clone has failed
-            string prepFileSystemError;
-            if (!GVFSPlatform.Instance.KernelDriver.TryPrepareFolderForCallbacks(enlistment.WorkingDirectoryRoot, out prepFileSystemError))
+            Result prepForCallbacksResult = new Result(true);
+            try
+            {
+                string prepFileSystemError;
+                if (!GVFSPlatform.Instance.KernelDriver.TryPrepareFolderForCallbacks(enlistment.WorkingDirectoryRoot, out prepFileSystemError))

```

Comment:
I think a better fix would be to have `TryPrepareFolderForCallbacks` handle its own exceptions - that's the contract of a TryFoo method after all, to convert exceptions to a false return value. And it would also enable us to leave the calling code here and in `DehydrateVerb` unchanged.

Reply:
@sanoursa I actually started with that approach (see the first commit here: https://github.com/Microsoft/VFSForGit/pull/280/commits/2c2d116711e613752d556378ecd3f9b3d5ca03e7), but I found that when testing the changes that the exception is happening when attempting to call `TryPrepareFolderForCallbacks` rather than when it attempts to call `ConvertDirectoryToVirtualizationRoot`.

My guess was that .NET must be validating that all dependencies for the method are available before the method is called, but I am not 100% sure.

Thinking about this more though I *could* do something like making `TryPrepareFolderForCallbacks` a wrapper for a helper method and catch the exceptions in `TryPrepareFolderForCallbacks`.  Thoughts on that approach?

Reply:
Yep the runtime validates all dependencies for a method at the time the method is called. Having `TryPrepareFolderForCallbacks` call another method to do the actual work is the best way to handle that type of error.